proposals/collection-expressions.md

```diff
@@ -188,23 +201,26 @@ namespace System.Runtime.CompilerServices
     }
 }
 ```
-
-The attribute can be applied to `class`, `struct`, and `interface` types.
+The attribute can be applied to a `class`, `struct`, `ref struct`, or `interface`.
 The attribute is not inherited although the attribute can be applied to a base `class` or an `abstract class`.
 
-An error is reported for a collection literal if there are multiple `[CollectionBuilder]` attributes on the collection type.
+The collection type must have an [*element type*](#element-type).
 
-The *create method* must be a `static` method defined on the *builder type*.
-The *create method* must have a single parameter of a type `System.ReadOnlySpan<T0>` passed by value, and a return value of the *collection type* to which the attribute is applied.
+For the *create method*:
+- The *builder type* must be a non-generic `class` or `struct`.
+- The method must be defined on the *builder type* directly.
+- The method must be `public` and `static`.
+- The *arity* of the method must match the *arity* of the collection type.
+- The method must have a single parameter of type `System.ReadOnlySpan<E>`, passed by value, where `E` is the [*element type*](#element-type) of the corresponding *collection type*.
+- The method return type must be the corresponding *collection type*.
```

Comment:
i still don't love this.  i would say that it either has to have a implicit-reference-conversion to the corresponding collection type.  Or just an implicit-conversion to the collection type.

It seems both unnecessarily restrictive *and* actually undesirable given that factory methods might return concrete types, but still be intended for other types (like interfaces).